skill-tree/src/main/java/com/RDS/skilltree/services/external/RdsServiceImplementation.java

```diff
@@ -0,0 +1,22 @@
+package com.RDS.skilltree.services.external;
+
+import com.RDS.skilltree.viewmodels.RdsUserViewModel;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+@RequiredArgsConstructor
+public class RdsServiceImplementation implements RdsService {
+    private final RestTemplate restTemplate;
+
+    @Value("${rds.backendBaseUrl}")
+    private String rdsBackendBaseUrl;
+
+    @Override
+    public RdsUserViewModel getUserDetails(String id) {
```

Comment:
can we wrap this into a try-catch block to handle exceptions and return appropriate error responses or log the issue for debugging.

Reply:
- added a try-catch block